test-case-reporting/app.ts

```diff
@@ -57,36 +68,42 @@ app.get('/', (req, res) => {
 });
 
 app.get('/test-results/build/:buildNumber', async (req, res) => {
-  // TODO(#949): Add error handling to GET endpoints
-  const {buildNumber} = req.params;
-  const {json} = req.query;
+  try {
+    const {buildNumber} = req.params;
+    const {json} = req.query;
 
-  const testRuns = await record.getTestRunsOfBuild(
-    buildNumber,
-    extractPageInfo(req)
-  );
+    const testRuns = await record.getTestRunsOfBuild(
+      buildNumber,
+      extractPageInfo(req)
+    );
 
-  if (json) {
-    res.json({testRuns});
-  } else {
-    res.send(renderTestRunList(testRuns));
+    if (json) {
+      res.json({testRuns});
+    } else {
+      res.send(renderTestRunList(testRuns));
+    }
+  } catch (error) {
+    handleError(error, res);
   }
 });
 
 app.get('/test-results/history/:testCaseId', async (req, res) => {
-  // TODO(#949): Add error handling to GET endpoints
-  const {testCaseId} = req.params;
-  const {json} = req.query;
+  try {
+    const {testCaseId} = req.params;
```

Comment:
nit: these can't throw errors, so let's move them out of the try/catch and have the try encompass as small a set of actions as possbile

Reply:
Done.